### PR TITLE
feat(perf): Add ability to put measurements directly on spans.

### PR DESCRIPTION
**Pull request by colin-sentry**
This functionality is already in other SDKs, and will be used in future AI features.

### CODE REVIEW

File: sentry_sdk/tracing.py [hunk 1]
```diff
@@ -155,6 +155,28 @@ def add(self, span):
             self.spans.append(span)
 
 
+class MeasurementValue:
```
Comment (Zylphrex) [inline sentry_sdk/tracing.py:158]: Thoughts on a typed dict instead of another class? This adds another layer to the serialization unnecessarily

Comment (colin-sentry) [reply sentry_sdk/tracing.py]: done

**Review by antonpirker — APPROVED** [review]
Yes, looks OK. 

But please do not merge for now. (Would make our live easier in case we need to fix something in SDK 2.0 that is discovered while dogfooding the SDK 2.0rc5 in sentry.io itself. We want to dog food until next week and then make the final 2.0 release)

**Comment by sentrivana** [comment]
Sorry for the wait folks! As 2.0 is out now, we're back to business so merging this!

**Pull request merged**